src/libraries/Microsoft.Extensions.Caching.Memory/tests/Microsoft.Extensions.Caching.Memory.Tests.csproj

```diff
@@ -10,7 +10,11 @@
              Link="Common\tests\Extensions\TestingUtils\Microsoft.AspNetCore.Testing\src\ExceptionAssertions.cs" />
     <Compile Include="..\..\Common\tests\Extensions\TestingUtils\Microsoft.AspNetCore.Testing\src\CultureReplacer.cs"
              Link="Common\tests\Extensions\TestingUtils\Microsoft.AspNetCore.Testing\src\CultureReplacer.cs" />
-    <ReferenceFromRuntime Include="Microsoft.Extensions.Caching.Memory" />
+  </ItemGroup>
+
+  <ItemGroup>
+    <ProjectReference Include="..\src\Microsoft.Extensions.Caching.Memory.csproj" SkipUseReferenceAssembly="true" />
```

Comment:
What does `SkipUseReferenceAssembly` do and why is it needed only here? I thought referencing the src project was like a `ReferenceFromRuntime`?

Reply:
The new model is that P2Ps always target the src projects (except for ref --> ref of course) but compile against the reference assembly that is produced in the ref project. This mimics the RefOut feature as you get the benefit of compiling against the ref assembly but get the src assembly as an app-local asset into the test app's output directory. Because of this we now don't need to store OOB assemblies in the testhost folder anymore and consume them the same way as customers do (app-local).

The `SkipUseReferenceAssembly` allows to compile against the src assembly which is the replacement for `ReferenceFromRuntime` but based on a P2P instead of a simple name reference.